### PR TITLE
Update Code and Fixed Some Warning ⚠

### PR DESCRIPTION
**Pull request by BillyFrcs**
## Overview
1. Fixed redundant code.
2. Fixed ``Possible System.InvalidOperationException`` warning.
3. Change some ``public`` method to ``private`` because it's not being use in the other class and ``protected`` as well.
4. Remove unused variables.
5. Remove virtual method that hasn't been override.
6. Fixed ``Possible System.NullReferenceException`` warning.

## Changes
- Fixes: Warning Code
- Fix ``Possible System.NullReferenceException`` warning.
- Fix ``Possible System.InvalidOperationException`` warning
- Fix redundant code

### CODE REVIEW

**Comment by BillyFrcs** [comment]
Why do the pipelines make some errors?

**Comment by timGerken** [comment]
@BillyFrcs , neither Unity 2018 nor Unity 2019 support switch pattern matching.

https://dev.azure.com/aipmr/MixedRealityToolkit-Unity-CI/_build/results?buildId=20950&view=logs&j=9a40d849-de61-58a5-b9fb-0035311f3ef5&t=40a1ca36-8c03-582a-2c45-de22c125c2c7&l=11115

**Comment by BillyFrcs** [comment]
Alright understood, Thanks for the correction @timGerken I've already revert it.

File: Assets/MRTK/SDK/Features/UX/Scripts/HandCoach/MoveToTarget.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ public class MoveToTarget : MonoBehaviour
         /// <summary>
         /// Object to track.
         /// </summary>
-        public GameObject TrackingObject
+        private GameObject TrackingObject
```
Comment (keveleigh) [inline Assets/MRTK/SDK/Features/UX/Scripts/HandCoach/MoveToTarget.cs:18]: Changing these from `public` to `private` is a breaking API change (which we try to avoid as much as possible), and I'm also not sure why they need to be changed. These are all helper scripts that could be called from anybody's application, even if MRTK doesn't call them directly.

**Review by keveleigh — CHANGES_REQUESTED** [review]

File: Assets/MRTK/SDK/Features/UX/Scripts/HandCoach/MoveToTarget.cs (hunk 1 above)
Comment (BillyFrcs) [reply Assets/MRTK/SDK/Features/UX/Scripts/HandCoach/MoveToTarget.cs]: Ah my bad, i didn't know that, thanks for the review, I've reverted it.

File: Assets/MRTK/SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs [hunk 2]
```diff
@@ -79,7 +79,7 @@ public abstract class BaseControllerPointer : ControllerPoseSynchronizer, IMixed
         /// </summary>
         /// <remarks>This <see href="https://docs.unity3d.com/ScriptReference/GameObject.html">GameObject</see> must have a <see cref="Microsoft.MixedReality.Toolkit.Input.IMixedRealityCursor"/> attached to it.</remarks>
         /// <param name="newCursor">The new cursor</param>
-        public virtual void SetCursor(GameObject newCursor = null)
+        private void SetCursor(GameObject newCursor = null)
```
Comment (RogPodge) [inline Assets/MRTK/SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs:82]: Hey, thanks for making a lot of these changes. I noticed that some formerly public functions are still private. Can you revert these changes, since they are technically breaking API changes?

This is just following up on Kurtis's observations from before.

**Review by RogPodge — CHANGES_REQUESTED** [review]

File: Assets/MRTK/SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs (hunk 2 above)
Comment (BillyFrcs) [reply Assets/MRTK/SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs:82]: Alright, thanks for the review!

**Comment by RogPodge** [comment]
Hey @BillyFrcs, I really appreciate you taking the time to address our feedback. We need all of the public fields that were changed to private/protected be reverted back to public, since they are breaking changes to our api.

We'd also like you to retarget this branch to prerelease/2.8.0 to get this in if you have the time.

**Comment by BillyFrcs** [comment]
> Hey @BillyFrcs, I really appreciate you taking the time to address our feedback. We need all of the public fields that were changed to private/protected be reverted back to public, since they are breaking changes to our api.
> 
> We'd also like you to retarget this branch to prerelease/2.8.0 to get this in if you have the time.

Okay, I'll try to take it back.

File: SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+<!-- BEGIN MICROSOFT SECURITY.MD V0.0.7 BLOCK -->
```
Comment (RogPodge) [inline SECURITY.md:1]: Where did this file come from?

**Review by RogPodge — CHANGES_REQUESTED** [review]

File: SECURITY.md (hunk 3 above)
Comment (keveleigh) [reply SECURITY.md:1]: This was automatically added to the main branch by an MSFT bot in #10599. It's not present in the 2.8 branch

File: Assets/MRTK/SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs [hunk 4]
```diff
@@ -245,15 +245,15 @@ protected override void OnDisable()
...
+        private bool isTeleportRequestActive = false;
 
         private bool lateRegisterTeleport = true;
 
         private bool canTeleport = false;
 
         private bool canMove = false;
 
-        protected Gradient GetLineGradient(TeleportSurfaceResult targetResult)
+        private Gradient GetLineGradient(TeleportSurfaceResult targetResult)
```
Comment (keveleigh) [inline Assets/MRTK/SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs:256]: Two more instances of visibility being lowered on scripts that might be overridden by consumers of MRTK

File: Assets/MRTK/SDK/Features/UX/Scripts/Pointers/ShellHandRayPointer.cs [hunk 5]
```diff
@@ -107,11 +107,15 @@ public override void OnPostSceneQuery()
                     wasGrabPressed = IsGrabPressed;
 
                     var currentMaterial = IsSelectPressed || IsGrabPressed ? lineMaterialSelected : lineMaterialNoTarget;
-
-                    for (int i = 0; i < LineRenderers.Length; i++)
+
```
Comment (keveleigh) [inline Assets/MRTK/SDK/Features/UX/Scripts/Pointers/ShellHandRayPointer.cs:110]:
```suggestion

```

File: Assets/MRTK/SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs [hunk 6]
```diff
@@ -79,7 +79,7 @@ public abstract class BaseControllerPointer : ControllerPoseSynchronizer, IMixed
         /// </summary>
         /// <remarks>This <see href="https://docs.unity3d.com/ScriptReference/GameObject.html">GameObject</see> must have a <see cref="Microsoft.MixedReality.Toolkit.Input.IMixedRealityCursor"/> attached to it.</remarks>
         /// <param name="newCursor">The new cursor</param>
-        public virtual void SetCursor(GameObject newCursor = null)
```
[the remark this answers is not in the document]
Comment (BillyFrcs) [reply Assets/MRTK/SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs:82]: > Does this compile? It looks like the return type is being removed but not replaced

Ah sorry, that was an accident.

File: Assets/MRTK/SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs (hunk 4 above)
Comment (BillyFrcs) [reply Assets/MRTK/SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs:256]: > Two more instances of visibility being lowered on scripts that might be overridden by consumers of MRTK

So, what sould i do?

File: SECURITY.md (hunk 3 above)
Comment (BillyFrcs) [reply SECURITY.md:1]: > Where did this file come from?

Um I dunno, I think it's auto generated when I push into this branch

File: Assets/MRTK/SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs (hunk 4 above)
Comment (RogPodge) [reply Assets/MRTK/SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs:256]: They should be left as protected to not lower the visibility of these variables

Comment (BillyFrcs) [reply Assets/MRTK/SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs:256]: Alright, understood.

File: Assets/MRTK/Providers/WindowsMixedReality/XRSDK/WindowsMixedRealityEyeGazeDataProvider.cs [hunk 7]
```diff
@@ -96,7 +97,7 @@ public override void Enable()
         private async void EnableIfLoaderBecomesActive()
         {
             await new WaitUntil(() => IsActiveLoader.HasValue);
-            if (IsActiveLoader.Value)
+            if (IsActiveLoader != null && IsActiveLoader.Value)
```
Comment (RogPodge) [inline Assets/MRTK/Providers/WindowsMixedReality/XRSDK/WindowsMixedRealityEyeGazeDataProvider.cs:100]: @keveleigh is the != null check necessary here?

File: Assets/MRTK/SDK/Features/UX/Scripts/Loader/LoaderController.cs [hunk 8]
```diff
@@ -196,10 +196,8 @@ private float Cycles
                 {
                     return (frequency < 0.0f) ? (Time.time / 0.000001f) * -1.0f : (Time.time / frequency) * -1.0f;
                 }
-                else
-                {
-                    return (frequency < 0.0f) ? Time.time / 0.000001f : Time.time / frequency;
-                }
+                
+                return (frequency < 0.0f) ? Time.time / 0.000001f : Time.time / frequency;
```
Comment (RogPodge) [inline Assets/MRTK/SDK/Features/UX/Scripts/Loader/LoaderController.cs:200]: I think this should get reverted back to the older pattern, since they should be equivalent functionally and the `else` helps make the code clearer.

**Comment by RogPodge** [comment]
Sorry for the delay on accepting this PR. Left some last comments, after which I think this PR should be good to go!

File: Assets/MRTK/SDK/Features/UX/Scripts/Loader/LoaderController.cs (hunk 8 above)
Comment (BillyFrcs) [reply Assets/MRTK/SDK/Features/UX/Scripts/Loader/LoaderController.cs:200]: Okay understood

**Comment by BillyFrcs** [comment]
> Sorry for the delay on accepting this PR. Left some last comments, after which I think this PR should be good to go!

Alright, no problem!
